File: src/de/saschahlusiak/freebloks/view/effects/Effect.java
package de.saschahlusiak.freebloks.view.effects;

import javax.microedition.khronos.opengles.GL10;
import javax.microedition.khronos.opengles.GL11;

import de.saschahlusiak.freebloks.view.BoardRenderer;

public interface Effect {
	boolean isEffected(int x, int y);
	void render(GL11 gl, BoardRenderer renderer);
	void renderShadow(GL11 gl, BoardRenderer renderer);
	boolean isDone();
	boolean execute(float elapsed);
}

File: src/main/java/de/saschahlusiak/freebloks/view/effects/Effect.kt
package de.saschahlusiak.freebloks.view.effects

import de.saschahlusiak.freebloks.view.BoardRenderer
import javax.microedition.khronos.opengles.GL11

interface Effect {
    fun isEffected(x: Int, y: Int): Boolean
    fun render(gl: GL11, renderer: BoardRenderer)
    fun renderShadow(gl: GL11, renderer: BoardRenderer)
    fun isDone(): Boolean
    fun execute(elapsed: Float): Boolean
}